bluebutton-data-pipeline-rif-extract/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/extract/s3/DataSetTestUtilities.java

```diff
@@ -160,7 +167,11 @@ public static PutObjectRequest createPutRequest(Bucket bucket, String keyPrefix,
 			return request;
 		} catch (IOException e) {
 			throw new UncheckedIOException(e);
+		} catch (NoSuchAlgorithmException e) {
+			throw new ChecksumException("NoSuchAlgorithmException on file " + manifest.getTimestampText()
+					+ manifestEntry.getName() + "trying to build md5chksum" + e);
```

Comment:
Ensure that your `ChecksumException` class can accept a `cause` parameter, and pass the causing exception through to the super constructor.
Otherwise, you risk losing stack trace info.